extensions/amp-form/0.1/form-proxy.js

```diff
@@ -109,3 +137,220 @@ function createFormProxyConstr(win) {
+ * for Chrome 45 and under.
+ *
+ * See https://developers.google.com/web/updates/2015/04/DOM-attributes-now-on-the-prototype-chain
+ * for more info.
+ *
+ * @param {!HTMLFormElement} form
+ * @param {!Object} proxy
+ */
+function setupLegacyProxy(form, proxy) {
+  const proto = form.cloneNode(/* deep */ false);
+  for (const name in proto) {
+    if (name in proxy ||
+        // Exclude constants.
+        name.toUpperCase() == name ||
+        // Exclude on-events.
+        name.substring(0, 2) == 'on') {
+      continue;
+    }
+    const desc = LEGACY_PROPS[name];
+    const current = form[name];
+    const isElement = (current && current.nodeType);
+    if (desc) {
+      if (desc.access == LegacyPropAccessType.READ_ONCE) {
+        // A property such as `style`. The only way is to read this value
+        // once and use it for all subsequent calls.
+        let actual;
+        if (isElement) {
+          // The overriding input, if present, has to be removed and re-added
+          // (renaming does NOT work).
+          const element = dev().assertElement(current);
+          const nextSibling = element.nextSibling;
+          const parent = element.parentNode;
+          parent.removeChild(element);
+          try {
+            actual = form[name];
+          } finally {
+            parent.insertBefore(element, nextSibling);
+          }
+        } else {
+          actual = current;
+        }
+        Object.defineProperty(proxy, name, {
+          get: function() {
+            return actual;
+          },
+        });
+      } else if (desc.access == LegacyPropAccessType.ATTR) {
+        // An attribute-based property. We can use DOM API to read and write
+        // with a minimal type conversion.
+        const attr = desc.attr || name;
+        Object.defineProperty(proxy, name, {
+          get: function() {
+            let value = proxy.getAttribute(attr);
+            if (value == null && desc.def !== undefined) {
+              value = desc.def;
+            } else if (desc.bool) {
+              if (desc.toggle) {
+                value = (value != null);
+              } else {
+                value = (value === 'true');
+              }
+            }
+            if (value && desc.url) {
+              value = parseUrl(/** @type {string} */ (value)).href;
+            }
+            return value;
+          },
+          set: function(value) {
+            if (desc.bool && desc.toggle) {
+              if (value) {
+                value = '';
+              } else {
+                value = null;
+              }
+            }
+            if (value != null) {
+              proxy.setAttribute(attr, value);
+            } else {
+              proxy.removeAttribute(attr);
+            }
+          },
+        });
+      } else {
+        dev().assert(false, 'unknown property access type: %s', desc.access);
+      }
+    } else {
+      // Not currently overriden by an input.
```

Comment:
I don't understand the comment. How does not having a `desc` mean there's no overriding input?